### PR TITLE
Simplify option parsing for Generator.Rewrite

### PR DESCRIPTION
**Pull request by Nihlus**
This PR replaces the homegrown CLI argument parsing in Generator.Rewrite with CommandLineParser in similar fashion to Generator.Convert.

Furthermore, it also adds some more reasonable error messages, and additional argument verification so that the program can bail out early.

### CODE REVIEW

File: src/Generator.Rewrite/Generator.Rewrite.csproj [hunk 1]
```diff
@@ -27,7 +27,8 @@
     <DocumentationFile>bin\Debug\Rewrite.xml</DocumentationFile>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <Commandlineparameters>../../OpenTK/Debug/OpenTK.dll ../../../OpenTK.snk -debug</Commandlineparameters>
+    <Commandlineparameters>--assembly ../../OpenTK/Debug/OpenTK.dll --signing-key ../../../OpenTK.snk --debug
+      -debug</Commandlineparameters>
```
Comment (Frassle) [inline src/Generator.Rewrite/Generator.Rewrite.csproj]: extra -debug?

File: src/Generator.Rewrite/Generator.Rewrite.csproj [hunk 2]
```diff
@@ -38,7 +39,8 @@
     <DocumentationFile>bin\Release\Rewrite.xml</DocumentationFile>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
-    <Commandlineparameters>../../OpenTK/Release/OpenTK.dll ../../../OpenTK.snk</Commandlineparameters>
+    <Commandlineparameters>--assembly ../../OpenTK/Release/OpenTK.dll --signing-key ../../../OpenTK
+      .snk</Commandlineparameters>
```
Comment (Frassle) [inline src/Generator.Rewrite/Generator.Rewrite.csproj]: odd line split

File: src/Generator.Rewrite/Options.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
+﻿using CommandLine;
+using CommandLine.Text;
+
+namespace OpenTK.Rewrite
+{
+    public class Options
+    {
+        [Option('i', "assembly", Required = true,
```
Comment (Frassle) [inline src/Generator.Rewrite/Options.cs]: I assume "i" for "input". Can it be consistent and be "a" though? I'd much sooner guess the first letter of "assembly" than I would "i" for input.

File: src/Generator.Rewrite/Program.cs [hunk 4]
```diff
@@ -301,19 +301,19 @@ private static void RemoveSupportingAttributes(TypeDefinition type)
                 generatedVariables = EmitConvenienceWrapper(wrapper, native, difference, body, il);
             }
 
-            if (slot != -1)
+            if (Options.UseDLLImport)
```
Comment (Frassle) [inline src/Generator.Rewrite/Program.cs]: Change in logic. Explanation? Looks to me like it should be if "slot == -1 || Options.UseDLLImport"

File: src/Generator.Rewrite/Options.cs (hunk 3 above)
Comment (daerogami) [reply src/Generator.Rewrite/Options.cs]: Do these have to be single letter identifiers?

Comment (Nihlus) [reply src/Generator.Rewrite/Options.cs]: @Frassle Yeah, that's reasonable. I'll change it.
@daerogami Yes, in line with the POSIX standard. You have a short identifier that is one character, and an optional longer identifier with an unlimited number of characters.

File: src/Generator.Rewrite/Program.cs (hunk 4 above)
Comment (Nihlus) [reply src/Generator.Rewrite/Program.cs]: Yes, you're absolutely correct.

**Comment by varon** [comment]
Thanks!

**Pull request merged**